tests/test_live_messages.py

```
@@ -0,0 +1,134 @@
+
+    with patch.object(nb, "get_bot_instance", return_value=mock_bot), patch.object(
+        nb, "get_async_session"
+    ) as mock_session_cls, patch.object(nb, "_bulk_fetch_matches", new_callable=AsyncMock) as mock_bulk_matches, patch.object(
+        nb, "_bulk_fetch_teams", new_callable=AsyncMock
+    ) as mock_bulk_teams, patch.object(nb, "_resolve_teams") as mock_resolve_teams, patch.object(
+        nb, "get_live_message_async", new_callable=AsyncMock
+    ) as mock_get_live, patch.object(nb, "set_live_message_async", new_callable=AsyncMock) as mock_set_live:
+        # Provide a guild config so per-guild channel resolution works in tests
+        with patch.object(
+            nb, "get_guild_config_async", new_callable=AsyncMock, return_value=MagicMock(live_updates_channel_id=channel.id, enabled_games=None)
+        ):
```

Ruff (0.15.5)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)